src/builder/value_parser.rs

```diff
@@ -936,15 +906,108 @@ impl Default for OsStringValueParser {
-#[derive(Copy, Clone, Debug)]
+#[derive(Clone)]
 #[non_exhaustive]
-pub struct PathBufValueParser {}
+pub struct PathBufValueParser {
+    canonicalize: bool,
+    starts_with: Option<String>,
+    ends_with: Option<String>,
+    exists: Option<bool>,
+    is_absolute: Option<bool>,
+    is_dir: Option<bool>,
+    is_file: Option<bool>,
+    is_relative: Option<bool>,
+    is_symlink: Option<bool>,
+}
+
+impl std::fmt::Debug for PathBufValueParser {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
+        f.debug_struct("PathBufValueParser").finish()
+    }
+}
+
+macro_rules! option_set_func {
+    ($t:ty, ($($name:tt),+)) => {
+        $(
+            /// Set the PathBufValueParser field
+            pub fn $name(mut self, value: $t) -> Self {
+                self.$name = Some(value);
+                self
+            }
+        )*
+    };
+}
+
+macro_rules! check_filebuf_property_with_param {
+    ($self:ident, $path:ident, $arg:ident, $cmd:ident, ($($name:tt),+)) => {
+        $(
+            if let Some($name) = &$self.$name {
+                if !$path.$name($name) {
+                    return Err(crate::Error::value_validation(
+                        $arg.map(ToString::to_string)
+                            .unwrap_or_else(|| "...".to_owned()),
+                        $path.to_string_lossy().to_string(),
+                        format!("{:?}.{}({:?}) failed", $path, stringify!($name), $name).into(),
+                    )
+                    .with_cmd($cmd));
+                }
+            }
+        )*
+    };
+}
+
+macro_rules! check_filebuf_property_eq {
+    ($self:ident, $path:ident, $arg:ident, $cmd:ident, ($($name:tt),+)) => {
+        $(
+            if let Some($name) = $self.$name {
+                if $path.$name() != $name {
+                    return Err(crate::Error::value_validation(
+                        $arg.map(ToString::to_string)
+                            .unwrap_or_else(|| "...".to_owned()),
+                        $path.to_string_lossy().to_string(),
+                        format!("{:?}.{}() == {:?} failed", $path, stringify!($name), $name).into(),
+                    )
+                    .with_cmd($cmd));
+                }
+            }
+        )*
+    };
+}
 
 impl PathBufValueParser {
     /// Implementation for [`ValueParser::path_buf`]
     pub fn new() -> Self {
-        Self {}
+        Self {
+            canonicalize: false,
+            starts_with: None,
+            ends_with: None,
+            exists: None,
+            is_absolute: None,
+            is_dir: None,
+            is_file: None,
+            is_relative: None,
+            is_symlink: None,
+        }
     }
+
+    /// Set flag to canonicalize the PathBuff
+    pub fn canonicalize(mut self) -> Self {
+        self.canonicalize = true;
+        self
+    }
+
+    option_set_func!(String, (starts_with, ends_with));
+    option_set_func!(
```

Comment:
I'd prefer we not use macros for all of the functions as it obscures what is happening